features/crc/soft/HalfByteCRC.h

```diff
@@ -0,0 +1,149 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_HALF_BYTE_CRC_H
+#define MBED_HALF_BYTE_CRC_H
```

Comment:
@geky - Please check if this implementation will suffice the CRC32 requirement for littlefs.
Tested with: https://gist.github.com/deepikabhavnani/ff6fc74c4150cff36d398c1d2a4e7f59

Reply:
Looks good to me! Only thought is that the name `HalfByteCRC` is a bit confusing. What about `MediumCRC`. Another thought, what if we renamed `SlowCRC` to `SmallCRC`. Just tossing out ideas, it's up to you.

other terribles names:
- `ModestCRC`
- `PracticalCRC`
- `KindaFastKindaSlowCRC`
- `BestCRC`

Reply:
Actually, thinking more about it. Maybe we should just leave out `HalfByteCRC` and force LittleFS to use `FastCRC`. It's probably better to have fewer options so there's more chances for the implementation to get reused.

We can also provide a config option for LittleFS to use `SlowCRC` (`SmallCRC`?) in the cases the size penalty is too much.

Thoughts?

Reply:
>  name HalfByteCRC is a bit confusing

Naming would be good based on speed/technique.
Since we have so many different variations  we can change from fast/slow to technique based.\
Fast CRC - `lookupTable` or `StandardTable` CRC
Slow CRC - `bitwiseCRC` or `tableless` CRC.
HalfByteCRC - Half-byte/Nibble/16-byte entry table are different options for

Reply:
> Actually, thinking more about it. Maybe we should just leave out HalfByteCRC and force LittleFS to use FastCRC

We can have default as FastCRC and switch to SlowCRC option in targets.json. Selection will be one-time mostly based on target and not application. Correct?
No harm in providing half-byte option as well, we just need to finalize one default.